### ISSUE TITLE
0.10-RC1: Incorrect example for TypesAnnotated.with()

### ISSUE THREAD

**Issue by davidburstrom** [ronmamo/reflections#337]
In the example in the [readme](https://github.com/ronmamo/reflections/blob/ref-0.10/README.md?plain=1#L38) it states that

```
Set<Class<?>> annotated = 
  reflections.get(TypesAnnotated.with(SomeAnnotation.class));
```

In 0.10-RC1, it results in a `Set<String>`, as `with()` is backed by `default QueryFunction<Store, String> with(Class<?>... keys) { return of(keys); }`. Not only that, the collection is empty contrary to what was returned in 0.9.12.

**Comment by ronmamo** [comment]
thanks @davidburstrom , I've fixed the readme example.
as for the empty result - shouldn't happen unless the scanner was configured, as tested in ReflectionsQueryTest. can you check the configuration or share how Reflections was instantiated?

**Comment by davidburstrom** [comment]
I'm a little surprised though, I'd have thought the README actually was more correct than the code.

What I used in 0.9.12 was 

```
  public static Set<Class<?>> getValidStuff() {
    return new Reflections("valid").getTypesAnnotatedWith(Stuff.class);
  }
```

which worked great.

In 0.10-RC1, I had to adapt the return type to

```
  public static Collection<Class<?>> getValidStuff() {
    return new Reflections("valid").getTypesAnnotatedWith(Stuff.class);
  }
```

which stopped returning results.

Further changing of that into

```
  public static Set<String> getValidStuff() {
    return new Reflections("valid").get(TypesAnnotated.with(Stuff.class));
  }
```

brought me even further from the target, and didn't return anything either.

**Comment by ronmamo** [comment]
will change the results from Collection back to Set.

empty result - suspected changes are in ConfigurationBuilder or FilterBuilder, can you check the differences in the effective `reflections.getConfiguration()`?

preferably use the explicit constructor to avoid ambiguity (will improve javadoc/readme), such as 
```
new Reflections(new ConfigurationBuilder().addUrls(ClasspathHelper.forPackage("valid")))
// or
new Reflections(new ConfigurationBuilder()
  .addUrls(ClasspathHelper.forPackage("valid")) 
  .setInputFilter(new FilterBuilder().includePattern(".*"))) // <-- optionally
```

either of the above will provide workable input, thanks for the coop!

**Comment by davidburstrom** [comment]
I personally have nothing against using `Collection` over `Set` in this case, but I suppose it does clarify that there will be no duplicates.

With 0.9.12 and

```java
new Reflections("valid").getTypesAnnotatedWith(OptionsInterface.class);
```

the configuration looks like

<img width="543" alt="0 9 12-configuration" src="https://user-images.githubusercontent.com/1671931/134806611-a5e14675-3828-47f0-a99a-fe9da5e206e3.png">

and with 0.10-RC1 and 

```java
    final ConfigurationBuilder configurationBuilder =
    new ConfigurationBuilder().addUrls(ClasspathHelper.forPackage("valid"));
    configurationBuilder.setInputsFilter(new FilterBuilder().includePattern(".*"));
    final Reflections reflections = new Reflections(configurationBuilder);
```

the configuration looks like

<img width="448" alt="0 10-rc1-configuration" src="https://user-images.githubusercontent.com/1671931/134806663-5bb698b7-30fb-417d-ae1c-b82eb1fd9836.png">

The URLs are identical and I've confirmed that they contain the expected classes.

I've yet to find a spec that produces any output.

**Comment by ronmamo** [comment]
thanks @davidburstrom for your valuable feedback and time! these will be addressed for the next release.

**Comment by davidburstrom** [comment]
Trying out RC2, there are no changes required to use the Reflections API, but still, when querying for all types annotated with X, there are zero results where there used to be non-zero with 0.9.12.

**Comment by ronmamo** [comment]
thanks @davidburstrom , but I couldn't find a way to reproduce it. it might be some javassist issue (downgrade can provide more input). you can send me the jar on email and i'll take a closer look.

**Comment by davidburstrom** [comment]
I'll try and reproduce it in a self-enclosed project, hang on.

**Comment by davidburstrom** [comment]
[repro-337.zip](https://github.com/ronmamo/reflections/files/7279633/repro-337.zip)

There you go, fails on my machine at least. Just run `gradle build`. I didn't provide a Gradle wrapper script.

**Comment by ronmamo** [comment]
ok got it. this is now marked as a known issue and will be fixed on next release, looks like it's going to be earlier than I thought...
as a workaround for 0.10, mark your annotation with `@Retention(RUNTIME)`

**Comment by davidburstrom** [comment]
Great, thanks!

**Comment by ronmamo** [comment]
fixed in 0.10.1

**Comment by davidburstrom** [comment]
I can confirm the fix works as intended in 0.10.1 :)
